R/import_metadata_and_counts.R

```diff
@@ -0,0 +1,118 @@
+#' Checks importing criteria for new atlas 
+#' Generats counts per million from provided raw counts
+#'
+#' @param sce_obj A SingleCellExperiment object from RDS
+#' @param cache_dir Optional character vector of length 1. A file path on
+#'   your local system to a directory (not a file) that will be used to store
+#'   `metadata.parquet`
+#' @export
+#' @return A metadata.parquet strip from SCE object. 
+#' Directories store counts and counts per million in the provided cache directory.
+#' @importFrom assertthat assert_that
+#' @importFrom checkmate check_tibble check_directory_exists check_set_equal check_true check_character check_subset check_file_exists
+#' @importFrom dplyr tbl select
+#' @importFrom cli cli_alert_info
+#' @importFrom glue glue
+#' @importFrom rlang .data
+#' @examples
+#' data(sample_sce_obj)
+#' import_metadata_counts(sample_sce_obj,
+#'                        cache_dir = get_default_cache_dir())
+import_metadata_counts <- function(
+  sce_obj,  
+  cache_dir = get_default_cache_dir()) {
+  original_dir <- file.path(cache_dir, "original")
+  
+  # Identify metadata and counts matrix
+  metadata_tbl <- sce_obj@metadata$data
+  counts_matrix <- sce_obj@assays@data$X
```

Comment:
You shouldn't access S4 slots directly if at all possible: https://adv-r.hadley.nz/s4.html#:~:text=Generally%2C%20you%20should,same%20external%20interface. 

I believe the methods you want here are:
* `S4Vectors::metadata()` instead of `sce_obj@metadata`
* `SummarizedExperiment::assay()` instead of `sce_obj@assays@data`